src/sentry/types/activity.py

```diff
@@ -0,0 +1,36 @@
+from enum import Enum
+
+
+class ActivityType(Enum):
+    SET_RESOLVED = 1
+    SET_UNRESOLVED = 2
+    SET_IGNORED = 3
+    SET_PUBLIC = 4
+    SET_PRIVATE = 5
+    SET_REGRESSION = 6
+    CREATE_ISSUE = 7
+    NOTE = 8
+    FIRST_SEEN = 9
+    RELEASE = 10
+    ASSIGNED = 11
+    UNASSIGNED = 12
+    SET_RESOLVED_IN_RELEASE = 13
+    MERGE = 14
+    SET_RESOLVED_BY_AGE = 15
+    SET_RESOLVED_IN_COMMIT = 16
+    DEPLOY = 17
+    NEW_PROCESSING_ISSUES = 18
+    UNMERGE_SOURCE = 19
+    UNMERGE_DESTINATION = 20
+    SET_RESOLVED_IN_PULL_REQUEST = 21
+
+    # The user has reprocessed the group, so events may have moved to new groups
+    REPROCESS = 22
+    MARK_REVIEWED = 23
+
+
+# Warning: This must remain in this EXACT order.
+CHOICES = tuple(
+    (i, ActivityType(i).name.lower())
+    for i in [1, 15, 13, 16, 21, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 14, 17, 18, 19, 20, 22, 23]
```

Comment:
I'm worried about this, is it really so bad to copy the `TYPE` from the other file instead?

Reply:
I think its worth it. The alternative gets pretty unreadable because of the enums:
```
SLUGS = {
    ActivityType.ASSIGNED: "assigned",
    ActivityType.CREATE_ISSUE: "create_issue",
    ActivityType.DEPLOY: "deploy",
    ActivityType.FIRST_SEEN: "first_seen",
    ActivityType.MARK_REVIEWED: "mark_reviewed",
    ActivityType.MERGE: "merge",
    ActivityType.NEW_PROCESSING_ISSUES: "new_processing_issues",
    ActivityType.NOTE: "note",
    ActivityType.RELEASE: "release",
    ActivityType.REPROCESS: "reprocess",
    ActivityType.SET_IGNORED: "set_ignored",
    ActivityType.SET_PRIVATE: "set_private",
    ActivityType.SET_PUBLIC: "set_public",
    ActivityType.SET_REGRESSION: "set_regression",
    ActivityType.SET_RESOLVED: "set_resolved",
    ActivityType.SET_RESOLVED_BY_AGE: "set_resolved_by_age",
    ActivityType.SET_RESOLVED_IN_COMMIT: "set_resolved_in_commit",
    ActivityType.SET_RESOLVED_IN_PULL_REQUEST: "set_resolved_in_pull_request",
    ActivityType.SET_RESOLVED_IN_RELEASE: "set_resolved_in_release",
    ActivityType.SET_UNRESOLVED: "set_unresolved",
    ActivityType.UNASSIGNED: "unassigned",
    ActivityType.UNMERGE_DESTINATION: "unmerge_destination",
    ActivityType.UNMERGE_SOURCE: "unmerge_source",
}

TYPE = (
    (ActivityType.SET_RESOLVED.value, SLUGS[ActivityType.SET_RESOLVED]),
    (ActivityType.SET_RESOLVED_BY_AGE.value, SLUGS[ActivityType.SET_RESOLVED_BY_AGE]),
    (ActivityType.SET_RESOLVED_IN_RELEASE.value, SLUGS[ActivityType.SET_RESOLVED_IN_RELEASE]),
    (ActivityType.SET_RESOLVED_IN_COMMIT.value, SLUGS[ActivityType.SET_RESOLVED_IN_COMMIT]),
    (ActivityType.SET_RESOLVED_IN_PULL_REQUEST.value, SLUGS[ActivityType.SET_RESOLVED_IN_PULL_REQUEST]),
    (ActivityType.SET_UNRESOLVED.value, SLUGS[ActivityType.SET_UNRESOLVED]),
    (ActivityType.SET_IGNORED.value, SLUGS[ActivityType.SET_IGNORED]),
    (ActivityType.SET_PUBLIC.value, SLUGS[ActivityType.SET_PUBLIC]),
    (ActivityType.SET_PRIVATE.value, SLUGS[ActivityType.SET_PRIVATE]),
    (ActivityType.SET_REGRESSION.value, SLUGS[ActivityType.SET_REGRESSION]),
    (ActivityType.CREATE_ISSUE.value, SLUGS[ActivityType.CREATE_ISSUE]),
    (ActivityType.NOTE.value, SLUGS[ActivityType.NOTE]),
    (ActivityType.FIRST_SEEN.value, SLUGS[ActivityType.FIRST_SEEN]),
    (ActivityType.RELEASE.value, SLUGS[ActivityType.RELEASE]),
    (ActivityType.ASSIGNED.value, SLUGS[ActivityType.ASSIGNED]),
    (ActivityType.UNASSIGNED.value, SLUGS[ActivityType.UNASSIGNED]),
    (ActivityType.MERGE.value, SLUGS[ActivityType.MERGE]),
    (ActivityType.DEPLOY.value, SLUGS[ActivityType.DEPLOY]),
    (ActivityType.NEW_PROCESSING_ISSUES.value, SLUGS[ActivityType.NEW_PROCESSING_ISSUES]),
    (ActivityType.UNMERGE_SOURCE.value, SLUGS[ActivityType.UNMERGE_SOURCE]),
    (ActivityType.UNMERGE_DESTINATION.value, SLUGS[ActivityType.UNMERGE_DESTINATION]),
    (ActivityType.REPROCESS.value, SLUGS[ActivityType.REPROCESS]),
    (ActivityType.MARK_REVIEWED.value, SLUGS[ActivityType.MARK_REVIEWED]),
)
```

Reply:
hmmm personally I prefer the long winded way (maybe a comment with the value would help?) but I'd be curious to see what others think here.